program: |
    fn main() : int {
      let a: int = 42;
    }

    fn a() : void {
    }

expect_compile_failure: true
expect_stderr: |
  2:7 Name error: Variable 'a' shadows a function
    let a: int = 42;
        ^
  5:4 Note: Previously declared here
  fn a() : void {
     ^
  Terminating compilation due to 1 error